sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretClientBuilder.java

```diff
@@ -150,19 +151,19 @@ public SecretAsyncClient buildAsyncClient() {
             .httpClient(httpClient)
             .build();
 
-        return new SecretAsyncClient(endpoint, pipeline, serviceVersion);
+        return new SecretAsyncClient(vaultUrl, pipeline, serviceVersion);
     }
 
     /**
      * Sets the vault endpoint url to send HTTP requests to.
      *
-     * @param endpoint The vault endpoint url is used as destination on Azure to send requests to.
+     * @param vaultUrl The vault endpoint url is used as destination on Azure to send requests to.
      * @return the updated {@link SecretClientBuilder} object.
-     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     * @throws IllegalArgumentException if {@code vaultUrl} is null or it cannot be parsed into a valid URL.
      */
-    public SecretClientBuilder endpoint(String endpoint) {
+    public SecretClientBuilder vaultUrl(String vaultUrl) {
         try {
-            this.endpoint = new URL(endpoint);
+            this.vaultUrl = new URL(vaultUrl);
         } catch (MalformedURLException e) {
             throw logger.logExceptionAsError(new IllegalArgumentException(
```

Comment:
Need to pass the exception object.
nit: In the logging messages do we still want to kepp `endpoint` ?

Reply:
updated